include/megaapi_impl.h

```diff
@@ -1275,7 +1276,7 @@ class MegaRequestPrivate : public MegaRequest
         MegaBackgroundMediaUpload* backgroundMediaUpload;  // non-owned pointer
 
     private:
-        unique_ptr<MegaBannerList> mBannerList;
+        unique_ptr<MegaBannerListPrivate> mBannerList;
```

Comment:
Why is it private instead of protected, as the rest of the members?
Also, to avoid casting issues, I think it should store a pointer to the base class :)

Reply:
> Why is it private instead of protected, as the rest of the members?

Probably just a habit, but now that I think of it, I can find a few good reasons:
- Because I could not find any reason for giving it more access -- it's more restrictive and still honors the purpose.
- Because it's an implementation detail, that is managed by a dedicated function that creates it, and accessed by another.
- Because an implementation detail should be allowed to change without affecting functionality -- think of creating a `MegaBannerListPrivate` on the fly;
- Because we can always blame the people that invented encapsulation.

So why not? I don't think "consistency" can be invoked here.

> Also, to avoid casting issues, I think it should store a pointer to the base class :)

I can't think of any issues with upcasting a _pointer_. Surely that's why they made it implicit. What am I missing?
Also, if I do that, then I will need to either explicitly cast it when used throughout the SDK, or add `add()` to the base class, which will make it modifiable by a user, which should not be allowed, right?

> Just picky comments, feel free to disregard those ones

I'll keep current code, as long as it's up to me. If you still want me to change them, please point them out again.

Reply:
:+1: good points for encapsulation. I asked because I thought I saw some others where you used `protected` by "default", but not here. I'm totally fine using private here, and in case a child class needs access, we can always change it :wink: 

As for the casting thing, if you haven't observed issue warnings or alike, it's all good. We definitely don't want to add `add()` to the public class.